sphinx/ext/autosummary/__init__.py

```diff
@@ -654,7 +656,9 @@ def _import_by_name(name: str) -> Tuple[Any, Any, str]:
             modname = '.'.join(name_parts[:j])
             try:
                 import_module(modname)
-            except ImportError:
+            except ImportError as e:
+                if last_errors is not None:
+                    last_errors.append(str(e))
```

Comment:
I think it would be better to store exception object itself instead of string. It's not too late to convert it to string on creating message.